src/highlight/highlight.ts

```diff
@@ -1,59 +1,35 @@
-import { Component, Input } from '@angular/core';
-import { bem, getPropertyByPath } from '../utils';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+import { Hit } from 'instantsearch.js';
+
+import { highlight } from 'instantsearch.js/es/helpers';
+import { getPropertyByPath } from 'instantsearch.js/es/lib/utils';
 
 @Component({
   selector: 'ais-highlight',
-  template: `<span [class]="cx()" [innerHtml]="content"></span>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<span class="ais-Highlight" [innerHtml]="content"></span>`,
 })
 export class NgAisHighlight {
   @Input() attribute: string;
-  @Input() hit: { _highlightResult?: {}; label?: string; highlighted?: string };
-  @Input() tagName: string = 'em';
-
-  cx = bem('Highlight');
+  @Input() hit: Partial<Hit>;
+  @Input() tagName: string = 'mark';
 
   get content() {
-    if (this.attribute === 'highlighted') {
-      return this.hit.highlighted
-        ? this.replaceWithTagName(this.hit.highlighted)
-        : this.hit.label;
-    }
-
-    if (this.hit.hasOwnProperty('_highlightResult')) {
-      const attributeHighlighted = getPropertyByPath(
-        this.hit._highlightResult,
-        this.attribute
-      );
-
-      // check that the attributeHighlighted is a string
-      if (
-        attributeHighlighted !== undefined &&
-        typeof attributeHighlighted.value === 'string'
-      ) {
-        return this.replaceWithTagName(attributeHighlighted.value);
-      }
-    }
-
+    const highlightAttributeResult = getPropertyByPath(
+      this.hit._highlightResult,
+      this.attribute
+    );
     const fallback = getPropertyByPath(this.hit, this.attribute);
-    if (!fallback) {
-      console.warn(
-        `Could not find attribute [${
-          this.attribute
-        }] into hit object, will display an empty string.`
-      );
 
-      return '';
+    if (!highlightAttributeResult && fallback) {
+      return fallback;
     }
 
-    return fallback;
-  }
-
-  replaceWithTagName(value: string) {
-    return value
-      .replace(
-        new RegExp('<em>', 'g'),
-        `<${this.tagName} class="${this.cx('highlighted')}">`
```

Comment:
this implies the default tag name is used, and it's em right? could you re-explain how mark isn't a breaking change?

Reply:
Yes, in my previous answer I also mixed things so I'll give you the full picture.

In the current implementation, `tagName` does not have an impact as its used in `replaceWithTagName`, which itself does not replace anything. This is because the `value` does not contain `<em>` but `<mark>` elements ([coming from InstantSearch.js](https://github.com/algolia/instantsearch.js/blob/dcbbd8804b4b6471d24820b42826b57388974c27/src/lib/utils/escape-highlight.ts#L5-L25)).

So in effect, a highlighted element would always be `... <mark>something</mark> ...`.

Now that the highlight is also computed from InstantSearch.js, we can provide it the `tagName` input and update its default to `<mark>`, in order to match with the default from IS.js.